examples/fuzzing/rt_n12/fuzz.py

```diff
@@ -0,0 +1,173 @@
+            pair = data.split(b'=')
+            key = str(pair[0].decode('utf-8')) # should not fail, unless NVRAM file is corrupt
+            if len(pair) != 2:
+                val = ""
+            else:
+                val = str(pair[1].decode('utf-8'))
+            self.nvram[key] = val
+
+    def my_nvram_set(self,ql:Qiling):
+        """
+        hook to emulate writing to NVRAM
+        """
+        value = self.ql.mem.string(ql.arch.regs.read("A1"))
+        key  = self.ql.mem.string(ql.arch.regs.read("A0"))
+       
+        self.nvram[value] = key
+        if self.dbg_level == QL_VERBOSE.DEBUG:
+            self.ql.log.info("Inside nvram set")
+            self.ql.log.info(value)
+            
+            self.ql.log.info(key)
+        self.fake_return()
+        return
+    def nvram_unset(self,ql:Qiling):
+        """
+        emulate clearing NVRAM
+        """
+        self.ql.log.info("fake unset")
+        self.fake_return()
+        return
+    def my_nvram_get_int(self, ql:Qiling):
+        """
+        hook emulating return an integer from NVRAM
+        """
+        self.ql.log.info("NVRAM get_int call")
+        key_addr = ql.arch.regs.read("A0")
+        key = str(ql.mem.string(key_addr))
+        val = self.nvram[key]
+        if val != '':
+            self.ql.arch.regs.write("V0", int(val, 16))
+        else:
+            self.ql.arch.regs.write("V0", 0x0)
+        self.fake_return()
+        return
+    def add_hooks(self):
+        """
+        hook all meaningful NVRAM calls
+        """
+        self.ql.hook_address(self.my_nvram_get, 0x00420690)
+        self.ql.hook_address(self.my_nvram_set, 0x004204b0)
+        self.ql.hook_address(self.my_nvram_get_int, 0x004201b0)
+        self.ql.hook_address(self.nvram_unset, 0x004206d0)
+    def __init__(self, dbg_level):
+        self.dbg_level = dbg_level
+        self.ql = Qiling(self.cmdline,rootfs=self.rootfs, verbose=dbg_level)
+        self.nvram = {}
+        self.nvram_addr = self.ql.mem.map_anywhere(size=4096, info='nvram label')
+        self.ql.fast_mode=True
+        self.populate_nvram()
+        self.add_hooks()
+def main():
+    parser = argparse.ArgumentParser(description='qiling example fuzzer for RT-N12 httpd binary')
+    parser.add_argument('--snapshot', action='store_true')
+    parser.add_argument('--fuzz', action='store_true')
+    parser.add_argument('--dbg', action='store_true')
+    parser.add_argument('--restore', action='store_true')
+    parser.add_argument('--filename', action='store')
+    parser.add_argument('--run', action='store_true')
+    args = parser.parse_args()
+    emu = Emulator(QL_VERBOSE.OFF)
+    if args.dbg:    
+        emu.ql.debugger="qdb"
+        
+    if args.restore:
+        emu.ql.restore(snapshot="httpd.bin")
+    if args.snapshot:
+        snapshot(emu.ql)
+    if args.run:
+        with cov_utils.collect_coverage(emu.ql, 'drcov', 'output.cov'):
+            emu.ql.run()
+    if args.fuzz and args.filename:
```

Comment:
It is not clear whether `run` and `fuzz` are mutually exclusive or not. If they are, then it needs to be reflected through arg parser.

Reply:
Fixed.